server/common/genesets.py

```diff
@@ -2,137 +2,8 @@
-                    continue
-                # if row starts with '#' it is a comment
-                if row[0].startswith("#"):
-                    continue
-                # if this is the first non-comment row, assume it is a header and validate
-                # column names.  OK if the user has extra columns after our initial set.
-                if not haveReadHeader:
-                    if row[0 : len(GENESETS_TIDYCSV_HEADER)] != GENESETS_TIDYCSV_HEADER:
-                        raise AnnotationsError("Gene set CSV file missing the required column header.")
-                    haveReadHeader = True
-                    continue
-
-                geneset_name, geneset_description, gene_symbol, gene_description, _ = just(5, row)
-                if not geneset_name:
-                    raise AnnotationsError(f"Gene set CSV missing required gene set name on line {lineno}")
-                if (not gene_symbol) and gene_description:
-                    messagefn(f"Warning: Missing gene name in gene set name {geneset_name} on line {lineno}.")
-
-                if geneset_name in gene_sets:
-                    gs = gene_sets[geneset_name]
-                else:
-                    gs = gene_sets[geneset_name] = {
-                        "geneset_name": geneset_name,
-                        "geneset_description": geneset_description,
-                        "genes": [],
-                    }
-                # Use first geneset_description with a value
-                if not gs["geneset_description"] and geneset_description:
-                    gs["geneset_description"] = geneset_description
-                # add the gene if the gene_symbol is defined
-                if gene_symbol:
-                    gs["genes"].append(
-                        {
-                            "gene_symbol": gene_symbol,
-                            "gene_description": gene_description,
-                        }
-                    )
-
-    return gene_sets
-
-
-def write_gene_sets_tidycsv(f, genesets):
-    """
-    Convert the internal gene sets format (returned by read_gene_set_tidycsv) into
-    the simple Tidy CSV.
-    """
-    writer = csv.writer(f, dialect="excel")
-    writer.writerow(GENESETS_TIDYCSV_HEADER)
-    for geneset in genesets:
-        # genes may be empty, in which case we skip the gene set entirely
-        genes = geneset["genes"]
-        if not genes:
-            writer.writerow([geneset["geneset_name"], geneset.get("geneset_description", ""), "", ""])
-        else:
-            writer.writerows(
-                [
-                    [
-                        geneset["geneset_name"],
-                        geneset.get("geneset_description", ""),
-                        gene["gene_symbol"],
-                        gene.get("gene_description", ""),
-                    ]
-                    for gene in genes
-                ]
-            )
+import re
 
 
 def summarizeQueryHash(raw_query):
```

Comment:
Is there a reason to keep this or `validate_gene_sets` below? I think we might be able to delete the file.

Reply:
It looks like `validate_gene_sets` isnt being used by `summarizeQueryHash` is called by `summarize_var_helper`.

Reply:
rm'd file! moved sole function to caller.